src/devtools/pachdev/actions.go

```diff
@@ -0,0 +1,571 @@
+	}
+
+	helmArgs := []string{"template", "pach", op.helmChartSource()}
+	helmArgs = append(helmArgs, op.getHelmArgs()...)
+	cmd := exec.Command("helm", helmArgs...)
+	stdout, stderr := &bytes.Buffer{}, &bytes.Buffer{}
+	cmd.Stdout, cmd.Stderr = stdout, stderr
+	if err := cmd.Run(); err != nil {
+		op.err = errors.Wrap(err, "failed to generate k8s manifest with helm:\n"+stderr.String())
+		return ""
+	}
+	op.k8sManifest = stdout.String()
+	return op.k8sManifest
+}
+
+// getImages returns a slice of all docker images that 'op' will reference in
+// it's Pachyderm deploy manifest *except* pachd and worker. These images can
+// downloaded (via 'docker pull') and pushed into the cluster (via 'kind load').
+func (op *deployOp) getImages() ([]string, error) {
+	if op.err != nil {
+		return nil, op.err
+	}
+
+	// Parse the helm chart (and minio k8s manifest, if in use) and extract all
+	// images in it. Unfortunately, while 'helm install' can write its output as
+	// JSON, 'helm template' cannot, so we query the output with yq.
+	decoder := yqlib.NewYamlDecoder(yqlib.NewDefaultYamlPreferences())
+	yamlDocs := []io.Reader{
+		strings.NewReader(op.getK8sManifest()),
+	}
+	if op.objectStorage == "minio" {
+		minioManifest, err := os.Open("etc/testing/minio.yaml")
+		if err != nil {
+			op.err = errors.Wrap(err, "could not read minio manifest to load minio image")
+			return nil, op.err
+		}
+		yamlDocs = append(yamlDocs, minioManifest)
+	}
+	images := []string{"jaegertracing/all-in-one:1.10.1"}
+	for _, doc := range yamlDocs {
+		reader := bufio.NewReader(doc)
+		if err := decoder.Init(reader); err != nil {
+			op.err = errors.Wrap(err, "could not initialize yqlib decoder")
+			return nil, op.err
+		}
+
+		for {
+			// decoder.Decode returns one yaml document, but the k8s manifest
+			// contains one document per k8s artifact (i.e. many documents
+			// total), so we'll need to decode all of them.
+			//
+			// Taken from https://github.com/mikefarah/yq/blob/aaef27147fadc1cc2d59c7f3c940f5281a1a8654/pkg/yqlib/all_at_once_evaluator_test.go#L35-L56
+			candidateNode, err := decoder.Decode()
+			if errors.Is(err, io.EOF) {
+				break
+			} else if err != nil {
+				op.err = errors.Wrap(err, "could not decode k8s manifest")
+				return nil, op.err
+			}
+			yqExpr := ".. | .image? | select(. | . != null)"
+			l, err := yqlib.NewAllAtOnceEvaluator().EvaluateNodes(yqExpr, candidateNode)
+			if err != nil {
+				op.err = errors.Wrapf(err, "could not evaluate yqlib expression %q against helm-generated manifest", yqExpr)
+				return nil, op.err
+			}
+			for el := l.Front(); el != nil; el = el.Next() {
+				image := el.Value.(*yqlib.CandidateNode).Value
+				if strings.Contains(image, "pachyderm/pachd") && op.pachVersion == "local" {
+					continue // this will be built and pushed in a different fn
+				}
+				images = append(images, image)
+			}
+		}
+	}
+
+	// sort & dedupe 'images' (wish there was a better way to do this)
+	slices.Sort(images)
```

Comment:
Why not store the required images in a map and sort.Strings(maps.Keys(images)) (where images is `map[string]struct{}`)?

As an aside, I don't think it would hurt new developers to have src/internal/sets that has a Set type ;)